### PR TITLE
Draft: UI ux 51 cards

### PR DESCRIPTION
**Pull request by MaddyRav**

### CODE REVIEW

File: frontend_new/src/App.tsx [hunk 1]
```diff
@@ -45,9 +46,11 @@ function App() {
           }}
         >
           <Grid item xs={12}>
+            {/* <MediaCard /> */}
```
Comment (EngineersBox) [inline frontend_new/src/App.tsx]: **[Required]** Lingering comment can be removed

File: frontend_new/src/themes/light.tsx [hunk 2]
```diff
@@ -3,13 +3,19 @@ import { createMuiTheme } from "@material-ui/core/styles";
 export const light = createMuiTheme({
   palette: {
     primary: {
-      main: "#f0f0f0",
+      // main: "#f0f0f0",
```
Comment (EngineersBox) [inline frontend_new/src/themes/light.tsx]: **[Required]** Lingering comments can be removed

File: frontend_new/src/components/Res.js [hunk 3]
```diff
@@ -19,19 +21,25 @@ function Res(props) {
...
-            <ListItem button>
-              <ListItemText
-                secondaryTypographyProps={{ style: textColor2 }}
-                primaryTypographyProps={{ style: textColor }}
-                primary={_source.title}
-                secondary={_source.summary}
+            <Grow in={true} timeout={time_delay}>
+
+            
+            <ListItem button onClick={() => console.log(_source.link)}>
```
Comment (EngineersBox) [inline frontend_new/src/components/Res.js:34]: **[Required]** We shouldn't be console logging anything, probably want to remove this.

File: frontend_new/src/components/Logo.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    return (
+      <Card className={classes.root}>
+        <CardMedia 
+            className={classes.root}
+            image= {require('logo.jpg')}
+            
+        />    
+    </Card>
+    );
+}
```
Comment (EngineersBox) [inline frontend_new/src/components/Logo.tsx:44]: **[Required]** Missing a newline at the end of this file

File: frontend_new/src/components/Res.js [hunk 5]
```diff
@@ -19,19 +21,25 @@ function Res(props) {
...
+            {() => setDelay(time_delay+200)}
```
Comment (EngineersBox) [inline frontend_new/src/components/Res.js]: **[Required, Maintainability, Magic Values]** Ideally, we should create a constant that describes exact what this value of `200` actually means. Having it inline could mean anything and it becomes a maintainability issue.

File: frontend_new/src/themes/light.tsx [hunk 6]
```diff
@@ -3,13 +3,19 @@ import { createMuiTheme } from "@material-ui/core/styles";
...
-      main: "#212121",
+      //main: "#212121",
+      main: "#F19828",
+      
     },
+    
+    
     background: {
-      default: "#dfdfdf",
+      default: "#f7f7f5",
```
Comment (EngineersBox) [inline frontend_new/src/themes/light.tsx]: **[Desired,Maintainability]** We should probably have a standalone file where we store all the global styling (main page elements). Having constants inline all over the code base is a maintainability nightmare.

File: frontend_new/src/components/Logo.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    },
+    pos: {
+      marginBottom: 12,
+    },
+    
+  });
+
+export default function MediaCard() {
+    const classes = useStyles();
+    const bull = <span className={classes.bullet}>•</span>;
```
Comment (EngineersBox) [inline frontend_new/src/components/Logo.tsx:33]: **[Desired, Compatibility]** There is no guarantee that the bullet point character will actually show up in all browsers. Probably a good idea to use a unicode character with HTML encoding instead.

File: frontend_new/src/App.tsx [hunk 8]
```diff
@@ -8,8 +8,9 @@ import Search from "./components/Search";
 import Grid from "@material-ui/core/Grid";
 import Typography from "@material-ui/core/Typography";
 import FrontPageInfo from "./components/FrontPageInfo";
-
 import Grow from "@material-ui/core/Grow";
+import MediaCard from "./components/Logo";
```
Comment (EngineersBox) [inline frontend_new/src/App.tsx]: **[Required]** Unused import, can be removed.

On a broader note, there probably don't need to be any changes to this file.

File: frontend_new/src/components/Cards.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  };
+  
+export default function SimpleCard(prop: props) {
+  const classes = useStyles();
+  const bull = <span className={classes.bullet}>•</span>;
+
+  return (
+    <Card className={classes.root}>
+      <CardContent style={{backgroundColor: "#F3F3F1"}}>
+        <Typography variant="h5" component="h2">
```
Comment (EngineersBox) [inline frontend_new/src/components/Cards.tsx:41]: **[Question, Partial API Usage]** Is there a reason that we are using a header with a raw `h2` component as opposed to the `CardHeader` component intended for cards? Seems like we should have subclassed styling for this particular variant of a card with the header if we want a custom header style.

File: frontend_new/src/components/FrontPageInfo.tsx [hunk 10]
```diff
@@ -14,16 +14,17 @@ export default function FrontPageInfo() {
       <Grid item xs={12}></Grid>
       <Grid item xs={12}>
         <Typography
-          color="secondary"
+          //color="secondary"
```
Comment (EngineersBox) [inline frontend_new/src/components/FrontPageInfo.tsx]: **[Required]** Commented out code can be removed

File: frontend_new/src/components/Res.js [hunk 11]
```diff
@@ -19,19 +21,25 @@ function Res(props) {
     color: "#9f9f9f",
   };
 
+  const [time_delay, setDelay]= useState(1);
+
   return (
     <List>
       {!resStatus &&
         res.result.result.map(({ _id, _score, _source }) => (
```
Comment (EngineersBox) [inline frontend_new/src/components/Res.js:30]: **[Question]** Is there a reason that we are using underscore prefixed entry mappings? Generally underscore prefixes indicate an unused field, but these are clearly used here.

File: frontend_new/src/components/Res.js [hunk 12]
```diff
@@ -4,6 +4,8 @@ import ListItemText from "@material-ui/core/ListItemText";
 import ListItem from "@material-ui/core/ListItem";
 import List from "@material-ui/core/List";
 import { useFetch } from "./Get";
+import SimpleCard from "./Cards.tsx";
+import Grow from '@material-ui/core/Grow';
 
 function Res(props) {
   const [res, resStatus] = useFetch(
```
Comment (EngineersBox) [inline frontend_new/src/components/Res.js:11]: **[Required, Maintainability]** Inline AJAX calls are a maintainability nightmare. We should be using services to mask API calls behind methods. The component itself shouldn't care about where the data comes from, it should only need to know what to call to get it. The service it references is designed to handle those specifics as an abstraction. Mixing abstraction layers is not a good idea and leads to over-coupling logic.

An article on consuming REST APIs with services in React: [https://dzone.com/articles/consuming-rest-api-with-reactjs](https://dzone.com/articles/consuming-rest-api-with-reactjs)

**Review by EngineersBox — CHANGES_REQUESTED** [review]
Nice progress, mostly some minor cleanup and addressing abstractive layering

File: frontend_new/src/components/Res.js (hunk 12 above)
Comment (jayjayjayjayjayjay) [reply frontend_new/src/components/Res.js:11]: This is an issue with UIUX-30 (currently the UI main branch), not the PR. It shouldn't be addressed in this PR.

File: frontend_new/src/components/Res.js (hunk 11 above)
Comment (jayjayjayjayjayjay) [reply frontend_new/src/components/Res.js:30]: This is an issue with UIUX-30 (currently the UI main branch), not the PR. It shouldn't be addressed in this PR.

File: frontend_new/src/components/Res.js (hunk 3 above)
Comment (jayjayjayjayjayjay) [reply frontend_new/src/components/Res.js:34]: This is going to be left in for use in a future ticket

**Pull request merged**